Swagger PR for Adding 2021-04-01-preview API version

mvvsubbu opened the pull request:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [x] new API version 
      - [ ] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet. April, 2021
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet. May, 2021
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [x] No, no need to refresh for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [x] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  -  [x] Ensure to copy the existing version into new directory structure for first commit (including refactoring) and then push new changes including version updates in separate commits. This is required to review the changes efficiently.
  - Adding a new service

- [x] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [x] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

mvvsubbu committed:
Adding baseline version from previous stable version

mvvsubbu committed:
Fixing Avocado errors

mvvsubbu committed:
Fixing API Version Mismatch Avacado error

jorgecotillo commented on specification/databricks/resource-manager/Microsoft.Databricks/preview/2021-04-01-preview/examples/DisableEncryption.json:
```diff
@@ -0,0 +1,130 @@
+    "subscriptionId": "subid",
+    "resourceGroupName": "rg",
+    "workspaceName": "myWorkspace",
+    "api-version": "2018-04-01",
```
>2018-04-01 [](start = 20, length = 10)

Make sure to update your api-version

mvvsubbu replied:
Fixed all the examples as well to use latest API version

mvvsubbu committed:
Updating SystemData

mvvsubbu committed:
Fixing examples to have latest API version

mvvsubbu committed:
Removing vNet Peering APIs as we aren't versioning them

mvvsubbu committed:
Fixing Lint Errors

mvvsubbu committed:
Updating CMK Properties

mvvsubbu committed:
Fixing lint errors

mvvsubbu commented on specification/databricks/resource-manager/Microsoft.Databricks/preview/2021-04-01-preview/databricks.json:
```diff
@@ -0,0 +1,1019 @@
+          "type": "string",
+          "description": "The encryption keySource (provider). Possible values (case-insensitive):  Microsoft.Keyvault",
+          "enum": [
+            "Microsoft.Keyvault"
```
Marking it as resolved, as this is not required in current API version scope planned the release.

mvvsubbu commented on specification/databricks/resource-manager/Microsoft.Databricks/preview/2021-04-01-preview/databricks.json:
```diff
@@ -0,0 +1,1019 @@
+          },
+          "required": [
+            "keyVaulturi",
+            "keyName"
```
We can enforce at server side and when this become optional we can remove on server side so not adding in swagger

jorgecotillo replied:
That's not a good approach, changing behavior on server side without letting users know is not good, if version is required for now, mark it here, once you relax the server side, you can create a new api-version this way users will know which api-version to use.

---
In reply to: [605851467](https://github.com/Azure/azure-rest-api-specs/pull/13736#discussion_r605851467) [](ancestors = 605851467)

mvvsubbu replied:
Fixed the swagger and made it mandatory

mvvsubbu commented on specification/databricks/resource-manager/Microsoft.Databricks/preview/2021-04-01-preview/databricks.json:
```diff
@@ -0,0 +1,1019 @@
+      },
+      "required": [
+        "keySource",
+        "keyVaultProperties"
```
I am making this as optional we can enforce it on server side logic

mvvsubbu commented on specification/databricks/resource-manager/Microsoft.Databricks/preview/2021-04-01-preview/databricks.json:
```diff
@@ -0,0 +1,1019 @@
+{
```
Great thanks for confirming it, so marking this one as resolved

mvvsubbu committed:
Fixed review comment

ashwnacharya reviewed (approved):
LGTM

mvvsubbu committed:
Fixing ARM Reviewer Comment

lirenhe commented:
/azp run

ashwnacharya commented on specification/databricks/resource-manager/Microsoft.Databricks/preview/2021-04-01-preview/databricks.json:
```diff
@@ -0,0 +1,1019 @@
+          },
+          "default": "Default"
+        },
+        "KeyName": {
```
Can you please fix the casing? it should be 'keyName`

ashwnacharya commented on specification/databricks/resource-manager/Microsoft.Databricks/preview/2021-04-01-preview/databricks.json:
```diff
@@ -0,0 +1,1019 @@
+          "description": "The version of KeyVault key.",
+          "x-ms-client-name": "KeyVersion"
+        },
+        "keyvaulturi": {
```
Can you please fix the casing? it should be 'keyVaultUri`

ashwnacharya commented on specification/databricks/resource-manager/Microsoft.Databricks/preview/2021-04-01-preview/databricks.json:
```diff
@@ -0,0 +1,1019 @@
+          "type": "string",
+          "description": "The name of KeyVault key."
+        },
+        "keyversion": {
```
Can you please fix the casing? it should be `keyVersion`

ashwnacharya commented on specification/databricks/resource-manager/Microsoft.Databricks/preview/2021-04-01-preview/databricks.json:
```diff
@@ -0,0 +1,1019 @@
+        },
+        "keyVaultProperties": {
+          "properties": {
+            "keyVaulturi": {
```
fix casing. it should be `keyVaultUri`

mvvsubbu replied:
Fixed the casing of the property

mvvsubbu committed:
Merge branch 'master' into users/subraman/cmknotebookfeature

mvvsubbu committed:
Bringing S360 fixes to new version

mvvsubbu committed:
Fixing the casing of the keyVaultUri

lirenhe commented:
@mvvsubbu, as discussed, please do the following to avoid cross version breaking change:
1. In 2018-04-01, split the APIs from 2 services into 2 json files.  
2. Update existing readme.md for the file change.   
3.  For tag package-2021-04-01-preview, please include 2021-04-01-preview/databricks.json and the json file of another service in previous API version

nishantpunetha committed:
adds private link properties, params and examples (#13986)

mvvsubbu committed:
Revert "adds private link properties, params and examples (#13986)"

This reverts commit 0498eaab0cdf6441acce6a85988ef06759751553.

mvvsubbu committed:
Fixing Cross Version Breaking Change Errors

mvvsubbu commented:
> @mvvsubbu, as discussed, please do the following to avoid cross version breaking change:
> 
> 1. In 2018-04-01, split the APIs from 2 services into 2 json files.
> 2. Update existing readme.md for the file change.
> 3. For tag package-2021-04-01-preview, please include 2021-04-01-preview/databricks.json and the json file of another service in previous API version

Fixed the comment

mvvsubbu commented:
@lirenhe I fixed the comments and still the CI checks are failing. I looked at them, they are tool related issues. Please review them and approve the PR. 

Once approved we want to take these changes to Private repo and merge there (Ideally we should have started there, given that we have started here want to complete approval here and using this approval want to merge in private repo), as the feature is still in private preview.

lirenhe commented:
/azp run

mvvsubbu committed:
Updating the vNet Peering into Preview Tag as well

lirenhe commented:
The breaking change is a false alarm as the operations are moved into another file.

zhenglaizhang commented:
/azp run

zhenglaizhang commented:
/azp run

zhenglaizhang commented:
/azp run

zhenglaizhang commented:
/azp run

mvvsubbu commented:
/azp run

zhenglaizhang commented:
/azp run

zhenglaizhang commented:
/azp run

mvvsubbu commented:
@lirenhe the PR is ready for merge, can you merge it?
